flutter/lib/src/sentry_native.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:meta/meta.dart';
+
+import '../sentry_flutter.dart';
+import 'sentry_native_channel.dart';
+
+/// [SentryNative] holds state that it fetches from to the native SDKs. Always
+/// use the shared instance with [SentryNative()].
+@internal
+class SentryNative {
+  SentryNative._();
+
+  static final SentryNative _instance = SentryNative._();
+
+  SentryNativeChannel? _nativeChannel;
+
+  factory SentryNative({SentryNativeChannel? nativeChannel}) {
+    if (nativeChannel != null) {
+      _instance._nativeChannel = nativeChannel;
+    }
+    return _instance;
+  }
```

Comment:
l: This for me is a "code smell" having something that looks like a constructor returning a shared instance, and you can even reconfigure it, which would affect it everywhere. 
I would rather have a static "shared" property.
I understand that dart has this concept of factory constructors, and singletons is one of the examples `factory` documentation suggests. But, only because we can, doesn't mean we should.

Reply:
GDEs for Flutter recommend this approach for Singletons, See https://twitter.com/vandadnp/status/1502860065270808582

The question for me here is, does it need to be a singleton? Looks like we can pass `SentryNative` instances via ctor to the needed integrations.

Reply:
> GDEs for Flutter recommend this approach for Singletons, See

I understand that. I just don't think this is a good thing to do. It's easier to make mistakes this way because the dev could think is getting a new object instead of an instance. 

But anyway, is just a personal preference. 

>The question for me here is, does it need to be a singleton? Looks like we can pass SentryNative instances via ctor to the needed integrations.

IMO it make sense to have classes that deal only with channel communication as singleton.

Reply:
| The question for me here is, does it need to be a singleton? Looks like we can pass SentryNative instances via ctor to the needed integrations.

We need some shared state, because we hold properties (e.g. `appStartEnd`) in this class, and we don't really have any other way to hold onto them except for options.

| But anyway, is just a personal preference.

This was also my take, I'm not super happy about having a singleton returned with `Class()` syntax, but also always preferred something like `Class.sharedInstance`, because you immediately see what it is. But if that's the way to go in `dart`, it's ok.

As for the `ctor` parameter, in essence I needed a way to provide the `Channel`, which is used internally, at a later time than instance creation, as it is only available to us when users call `init`.

We could at least move this injection to a method to make it more explicit. WDYT?

Reply:
`SentryNative` needs to be used in the `SentryNavigatorObserver`, and `SentryNavigatorObserver` is created by the user, so it needs to be a singleton/shared instance.
Another way would be if `SentryNavigatorObserver` has access to the MethodChannel, then it could pass the instance per method call for `SentryNative`, but not sure if it makes much difference.
@denrase your call.

Reply:
@marandaneto Then we'd somehow need to provide the channel instance to the SentryNavigatorObserver, but it is a static in `SentryFlutter` and can also be overridden in the init method.

@brustolin We now use method injection for the channel dependency, so the intent is clear and there's no unneeded additional confusion with the `ctor` call.